### PR TITLE
Hy auth

### PR DESCRIPTION
**Pull request by HristoYonkov**
1. Change the form validations and inputs
2. Update logic for logout and now it is working with form properly
3. We need to check the logout function

### CODE REVIEW

File: src/contexts/Auth.tsx [hunk 1]
```diff
@@ -26,16 +25,20 @@ export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
...
-      if (!accessToken) {
-        return;
-      }
       try {
-        const response = await axiosInstance.get('/auth/me');
-        setUserState({username: response.data.username, role: response.data.role});
+        const response = await getWarehouseManagementApi().getApiUserMe()
+        
+        const username = response.userName ?? null;
+        const role = (response.roles && response.roles.length > 0) ? response.roles[0] : null;
```
Comment (vexleet) [inline src/contexts/Auth.tsx]: I added a new property inside `GET /me` which returns just the role so you can generate types and use it.
image

File: src/hooks/services/auth/useAuth.ts [hunk 2]
```diff
@@ -1,81 +1,57 @@
...
-      setUser({ username: requestedUser.username, role: requestedUser.role });
-      return { username: requestedUser.username, role: requestedUser.role };
+      const requestedUser = await getCurrentLoggedUser()
+
+      const username = requestedUser?.userName ?? null;
+      const role = requestedUser?.roles?.[0] ?? null;
+
+      const user = username && role ? { username, role } : null;
+
+      setUser(user);
```
Comment (vexleet) [inline src/hooks/services/auth/useAuth.ts]: you will always have userName and a role so just call `setUser({ username: requestedUser.username, role: requestedUser.role })` and you can add some checks in there. Here, you shouldnt do that, you just need to login and set the user.

File: src/hooks/services/auth/useAuth.ts [hunk 3]
```diff
@@ -1,81 +1,57 @@
...
     }
     return null;
   }
 
   const logoutUser = async () => {
     try {
-      // logout from BE
-      removeTokens();
+      // TODO: check the logout..
+      // const response = await getWarehouseManagementApi().postApiAuthLogout()
```
Comment (vexleet) [inline src/hooks/services/auth/useAuth.ts]: uncomment this and just use `await getWarehouseManagementApi().postApiAuthLogout()`

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -24,8 +24,10 @@ export default function Login() {
 
   const onSubmit: SubmitHandler<LoginFormData> = async (data) => {
     console.log(data);
+    // username: Ico
+    // password: Asd1!
```
Comment (vexleet) [inline src/pages/Login.tsx]: delete

File: src/services/api.ts [hunk 5]
```diff
@@ -1,22 +1,53 @@
...
-  baseURL: 'https://leads-academy-intercars.com'
+  baseURL: 'https://leads-academy-intercars.com',
+  withCredentials: true
 })
 
+const refreshToken = async (): Promise<void> => {
+  try {
+    await axios.post('https://leads-academy-intercars.com/api/Auth/refresh', {}, {
+      withCredentials: true,  // Ensure cookies are sent with the request
+    });
```
Comment (vexleet) [inline src/services/api.ts]: why you use `axios.post` and not the generated API?

File: package.json
Comment (vexleet) [inline package.json:1]: we have `js-cookie` package in here. Remove it

File: src/services/api.ts (hunk 5 above)
Comment (HristoYonkov) [reply src/services/api.ts]: I thought it would become a circular dependency?

File: src/contexts/Auth.tsx (hunk 1 above)
Comment (HristoYonkov) [reply src/contexts/Auth.tsx]: change the logic, check now

File: src/hooks/services/auth/useAuth.ts (hunk 2 above)
Comment (HristoYonkov) [reply src/hooks/services/auth/useAuth.ts]: check now Kriss

**Review by vexleet — APPROVED** [review]

**Pull request merged**